Trimble.Modus.Components/Controls/TMRadioButton/TMRadioButtonGroup.cs

```diff
@@ -0,0 +1,501 @@
+    {
+        int index = 0;
+        LoopChildren(_buttonContainer.Children, index);
+    }
+
+    private void SetDefaultSelectedRadioButton()
+    {
+        if (SelectedIndex < 0 || SelectedIndex >= _radioButtons.Count)
+        {
+            return;
+        }
+
+        _radioButtons[SelectedIndex].IsSelected = true;
+    }
+
+    /// <summary>
+    /// Loops through the children of the <see cref="TMRadioButtonGroup"/> and sets the <see cref="TMRadioButton.IsSelected"/> property of the radio buttons in the group
+    /// </summary>
+    /// <param name="children">The children of the <see cref="TMRadioButtonGroup"/></param>
+    protected override void OnChildAdded(Element child)
+    {
+        base.OnChildAdded(child);
+        if (child is TMRadioButton radioButton)
+        {
+            this.Children.Remove(radioButton);
+            AddRadioButtons(radioButton);
+        }
+        else if (this.Children.Count > 2 && child is not TMRadioButton)
+        {
+            throw new ArgumentException($"Only {nameof(TMRadioButton)} can be added to {nameof(TMRadioButtonGroup)}");
+        }
+    }
+
+    /// <summary>
+    /// Adds a <see cref="TMRadioButton"/> to the list of radio buttons.
+    /// </summary>
+    /// <param name="radioButton">The <see cref="TMRadioButton"/> to add.</param>
+    private void AddRadioButtons(TMRadioButton radioButton)
+    {
+        _radioButtons.Add(radioButton);
+        _buttonContainer.Children.Add(radioButton);
+        if (_radioButtons.Count == SelectedIndex + 1)
+        {
+            radioButton.IsSelected = true;
+        }
+        radioButton.IsEnabled = IsEnabled;
+        radioButton.Size = Size;
+        radioButton.SelectionChanged += RadioButtonSelectionChanged;
+    }
+
+    /// <summary>
+    /// This method is called when a radio button is selected.
+    /// </summary>
+    /// <param name="sender"></param>
+    /// <param name="e"></param>
+    private void RadioButtonSelectionChanged(object sender, EventArgs e)
+    {
+        if (sender is not TMRadioButton selectedRadioButton || !selectedRadioButton.IsSelected)
+        {
+            return;
+        }
+
+        foreach (var radioButton in _radioButtons)
+        {
+            if (radioButton != selectedRadioButton)
+            {
+                radioButton.IsSelected = false;
+            }
+        }
+
+        int index = FindIndex(selectedRadioButton);
+        SelectedIndex = index;
+        object value = selectedRadioButton.Value;
+        var eventArgs = new TMRadioButtonEventArgs(value, index);
+        SelectedRadioButtonChangedCommand?.Execute(eventArgs);
+        SelectedRadioButtonChanged?.Invoke(this, eventArgs);
+    }
+
+    /// <summary>
+    /// Finds the index of the selected radio button in the group
+    /// </summary>
+    /// <param name="radioButton">Selected Radio button</param>
+    /// <returns>integer index</returns>
+    private int FindIndex(TMRadioButton radioButton)
```

Comment:
Do we need this method and the `FindIndexRecursive` now? As we will have only the `TMRadioButton` as the children, can't we use just the `FindIndex(x => x.Text == buttonTitle);`?

Reply:
Not needed. Changed to `_radioButtons.IndexOf(selectedRadioButton);`